File: 34. Find First and Last Position of Element in Sorted Array/34_test.go
package leetcode

import(
	"testing"
	"reflect"
)

func TestSearchRange(t *testing.T){
	type test struct{
		input1 []int
		input2 int
		want []int
	}

	tests := map[string]test{
		"1": test{[]int{1}, 1, []int{0,0}},
		"2": test{[]int{5,7,7,8,8,10}, 8, []int{3,4}},
		"3": test{[]int{5,7,7,8,8,10}, 6, []int{-1,-1}},
	}

	for name, tc := range tests{
		t.Run(name, func(t *testing.T){
			got := searchRange(tc.input1, tc.input2)
			if !reflect.DeepEqual(got, tc.want){
				t.Errorf("Test Failed! want: %#v got: %#v", tc.want, got)
			}
		})
	}
}

func BenchmarkSearchRange(b *testing.B){
	for i:=0; i<b.N; i++{
		searchRange([]int{5,7,7,8,8,10}, 8)
	}
}

File: 34. Find First and Last Position of Element in Sorted Array/_34_test.py
import unittest
from _34 import *

class test():
    def __init__(self, input1, input2, want):
        self.input1 = input1
        self.input2 = input2
        self.want = want

class myTest(unittest.TestCase):
    def testSearchRange(self):
        so = Solution()
        test1 = test([1],1,[0,0])
        test2 = test([5,7,7,8,8,10],8,[3,4])
        test3 = test([5,7,7,8,8,10],6,[-1,-1])

        want1 = so.searchRange(test1.input1, test1.input2)
        want2 = so.searchRange(test2.input1, test2.input2)
        want3 = so.searchRange(test3.input1, test3.input2)

        self.assertEqual(want1, test1.want, "Test1 Failed!")
        self.assertEqual(want2, test2.want, "Test2 Failed!")
        self.assertEqual(want3, test3.want, "Test3 Failed!")

if __name__ == "__main__":
    testSuite = unittest.TestSuite()
    testSuite.addTest(myTest("testSearchRange"))
    unittest.TextTestRunner(testSuite)
